contracts/toCKB-typescript/src/switch/auction_signertimeout/mod.rs

```diff
@@ -1,7 +1,218 @@
+            eth_lot_size.get_sudt_amount()
+        }
+    };
+
+    debug!("begin verify since");
+    let auction_time = verify_since()?;
+    debug!("begin verify input");
+    verify_inputs(toCKB_lock_hash.as_ref(), lot_amount)?;
+    debug!("begin verify output");
+    verify_outputs(
+        input_data,
+        auction_time,
+        toCKB_lock_hash.as_ref(),
+        lot_amount,
+    )?;
+
+    Ok(())
+}
+
+fn verify_since() -> Result<u64, Error> {
+    let since = load_input_since(0, Source::GroupInput).map_err(|_| Error::InputSinceInvalid)?;
+
+    if since & REMAIN_FLAGS_BITS != 0 // check flags is valid
+        || since & LOCK_TYPE_FLAG == 0 // check if it is relative_flag
+        || since & METRIC_TYPE_FLAG_MASK != SINCE_TYPE_TIMESTAMP
+    // check if it is timestamp value
+    {
+        return Err(Error::InputSinceInvalid);
+    }
+
+    let auction_time = since & VALUE_MASK;
+    Ok(auction_time)
+}
+
+fn verify_inputs(toCKB_lock_hash: &[u8], lot_amount: u128) -> Result<(), Error> {
+    // check XT cell on inputs
+    let mut input_index = 1;
+    let mut sum_amount = 0;
+    loop {
+        let res = load_cell_type(input_index, Source::Input);
+        if res.is_err() {
+            break;
+        }
+        let script = res.unwrap();
+        if script.is_none() || !is_XT_typescript(&script.unwrap(), toCKB_lock_hash) {
+            return Err(Error::InvalidInputs);
+        }
+
+        let cell_data = load_cell_data(input_index, Source::Input)?;
+        let mut data = [0u8; UDT_LEN];
+        data.copy_from_slice(&cell_data);
+        let amount = u128::from_le_bytes(data);
+        sum_amount += amount;
+
+        input_index += 1;
+    }
+
+    if sum_amount < lot_amount {
+        return Err(Error::FundingNotEnough);
+    }
+    Ok(())
+}
+
+fn verify_outputs(
+    input_data: &ToCKBCellDataView,
+    auction_time: u64,
+    toCKB_lock_hash: &[u8],
+    lot_amount: u128,
+) -> Result<(), Error> {
+    // check bidder cell
+    debug!("begin check bidder cell");
+    debug!(
+        "auction_time: {}, AUCTION_MAX_TIME: {}",
+        auction_time, AUCTION_MAX_TIME
+    );
+
+    let mut output_index = 0;
+    // - 1. check bidder lock
+    if load_cell_lock_hash(output_index, Source::Output)? != load_cell_lock_hash(1, Source::Input)?
+    {
+        return Err(Error::InvalidAuctionBidderCell);
+    }
+    debug!("1. check bidder lock success! ");
+
+    // expect paying ckb to bidder,trigger and signer
+    // cap of toCKB_cell ==  XT_CELL_CAPACITY + to_bidder + to_trigger + to_signer
+    let collateral = load_cell_capacity(0, Source::GroupInput)? - XT_CELL_CAPACITY;
+    let mut to_bidder = collateral;
+    let init_collateral = collateral * AUCTION_INIT_PERCENT as u64 / 100;
+    if auction_time == 0 {
+        to_bidder = init_collateral
+    } else if auction_time < AUCTION_MAX_TIME {
+        to_bidder =
+            init_collateral + (collateral - init_collateral) * auction_time / AUCTION_MAX_TIME
+    }
```

Comment:
```suggestion
    if auction_time < AUCTION_MAX_TIME {
        to_bidder =
            init_collateral + (collateral - init_collateral) * auction_time / AUCTION_MAX_TIME
    }
```

Reply:
if auction_time == 0, does to_bidder == collateral ?. In my opinion, it should == init_collateral that is 67% * collateral